src/carousel/carousel.controller.ts

```diff
@@ -62,24 +66,51 @@ export class CarouselController {
     return toCarouselSlideDto(carouselSlide);
   }
 
-  private throttledFindAll = throttle(
-    async () => {
-      const carouselSlides = await this.carouselService.findAll();
+  /* Add a new carousel slide translation */
+  @Roles(Role.Superadmin)
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({ type: CreateCarouselSlideTranslationSwaggerDto })
+  @UseInterceptors(FileFieldsInterceptor([{ name: 'image', maxCount: 1 }]))
+  @Post('slides/:id/translation/add')
+  async addTranslation(
+    @Param('id') id: number,
+    @Body()
+    createCarouselSlideTranslationDto: CreateCarouselSlideTranslationDto,
+    @UploadedFiles({
+      transform: (val) => plainToInstance(CreateCarouselSlideFilesDto, val),
+    })
+    files: CreateCarouselSlideFilesDto,
+  ) {
+    await this.carouselService.addTranslation(
+      +id,
+      createCarouselSlideTranslationDto,
+      files,
+    );
+  }
+
+  private throttledFindAll = memoizeThrottle(
+    async (language: Language) => {
+      const carouselSlides = await this.carouselService.findAll(language);
       return toCarouselSlideDtoArray(carouselSlides);
     },
     24 * 60 * 60 * 1000, // 24 hours
   );
 
   /* Get all carousel slides */
   @Get('slides/get')
-  async findAll() {
-    return await this.throttledFindAll();
+  async findAll(@Query() query: LanguageDto) {
+    const language = query.lang ?? Language.En;
+    return this.throttledFindAll(language);
   }
 
   /* Get specific carousel slide by unique id */
   @Get('slides/get/:id')
-  async findOne(@Param('id') id: string): Promise<CarouselSlideDto> {
-    const carouselSlide = await this.carouselService.findOne(+id);
+  async findOne(
+    @Param('id') id: string,
+    @Query() query: LanguageDto,
+  ): Promise<CarouselSlideDto> {
+    const language = query.lang ?? Language.En;
```

Comment:
you can add this fallback Language.English on the line 109 like so: `@Param('id') id = Language.English`